test/driver.cpp

```diff
@@ -92,46 +97,52 @@ void processCmdLine(std::vector<std::string_view> arguments)
    //extract the exe name without path or filename extension
    //the result is used to expand the macro $exe
    std::filesystem::path exePath(arguments[0]);
-   std::string filenameNoExtension = exePath.replace_extension("").filename().string();
+   string filenameNoExt = exePath.replace_extension("").filename().string();
 
-   //replace $exe macro in header text
+   //replace $exe macro in header text only if a header text was defined
+   //and the print header option is enabled
    if (printHeader && !headerText.empty())
    {
-      std::string::size_type macroPosition = headerText.find("$exe");
-      for (; macroPosition != std::string::npos; macroPosition = headerText.find("$exe"))
-         headerText.replace(macroPosition, 4, filenameNoExtension);
+      string::size_type pos = headerText.find("$exe");
+      for (; pos != string::npos; pos = headerText.find("$exe"))
+         headerText.replace(pos, 4, filenameNoExt);
    }
    else
       headerText = "";
 
    setHeaderText(headerText);
 
-   //replace $exe macro in filename
+   //if a filename was supplied, replace $exe macro in filename
    if (!outputFilename.empty())
    {
-      std::string::size_type macroPosition = outputFilename.find("$exe");
-      for (; macroPosition != std::string::npos; macroPosition = outputFilename.find("$exe"))
-         outputFilename.replace(macroPosition, 4, filenameNoExtension);
+      string::size_type pos = outputFilename.find("$exe");
+      for (; pos != string::npos; pos = outputFilename.find("$exe"))
+         outputFilename.replace(pos, 4, filenameNoExt);
```

Comment:
The code to expand the `$exe` macro occurs twice. Following the DRY principle (Don't Repeat Yourself), I recommend refactoring the repeated code to a function with a declaration such as the following. A careful review of the `std::string` docs could also help, but I don't have much insight to add on that at this very moment.

 `void replace_all(std::string& str, const std::string& substr);`

Reply:
Thank you for the feedback @smurthys, I agree that the repeated code should be refactored. Is there any reason the `substr` parameter can't be a `string_view` since it will only be read?

Reply:
Yes @Ndemco, the second parameter could be `string_view`. Also, I notice the declaration I provided is missing a parameter. I do advise writing the function in isolation and verify that it works before changing the larger code:

`void replace_all(std::string& str, const std::string_view& substr, const std::string_view replacement);`

Reply:
@smurthys I've made the changes and pushed to origin. For `replace_all()` I used a `const string&` as the `toReplace` parameter since the text we want to replace the substring with (`filenameNoExt`) already exists as a string. I did try to change `filenameNoExt` to a `string_view` and then change the `toReplace` parameter to a `const string_view&` to match, but it caused bugs. For some reason when `filenameNoExt` is a `string_view`, its value is missing the first letter of the expected text; "test-stl-lite" becomes "est-stl-lite". For now I have left the variable as a `string` and the parameter as a `const string&` until I can investigate further.